dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/table/JdbcEventAnalyticsTableManager.java

```diff
@@ -157,7 +157,11 @@ public AnalyticsTableType getAnalyticsTableType()
 
         Calendar calendar = PeriodType.getCalendar();
 
-        List<Program> programs = idObjectManager.getAllNoAcl( Program.class );
+        List<Program> programs = params.getSkipPrograms().size() == 0 ? idObjectManager.getAllNoAcl( Program.class )
```

Comment:
I think we can add a method `hasSkipPrograms` on the params object to encapsulate the logic and remove the size == 0 check in this class.

Reply:
Done